frappe/email/doctype/auto_email_report/auto_email_report.py

```diff
@@ -16,8 +16,10 @@ class AutoEmailReport(Document):
 	def autoname(self):
 		self.name = _(self.report)
 
-	def validate(self):
+	def before_insert(self):
 		self.validate_report_count()
+
+	def validate(self):
 		self.validate_emails()
```

Comment:
why not in validate? If it is only added on insert, it will lead to a work-around?